src/app/loans/add-collateral/add-collateral.component.ts

```diff
@@ -0,0 +1,74 @@
+/** Angular Imports. */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+/** Custom Services. */
+import { LoansService } from '../loans.service';
+
+/**
+ * Add Collateral component.
+ */
+@Component({
+  selector: 'mifosx-add-collateral',
+  templateUrl: './add-collateral.component.html',
+  styleUrls: ['./add-collateral.component.scss']
+})
+export class AddCollateralComponent implements OnInit {
+
+  /** Collateral form. */
+  collateralForm: FormGroup;
+  /** Collateral data. */
+  collateralData: any;
+  /** Loan Id. */
+  loanId: string;
+
+  /**
+   * Retrieve data from `Resolver`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {Router} router Router.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {LoansService} LoansService loans service.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private router: Router,
+              private route: ActivatedRoute,
+              private loanService: LoansService ) {
+    this.route.data.subscribe((data: {collateralData: any, loanData: any}) => {
+      this.collateralData = data.collateralData;
+      this.loanId = data.loanData.id;
```

Comment:
```suggestion
      this.loanId = this.route.parent.snapshot.params['loanId'];
```
You don't need to call the API to get the loan id.